<?xml version="1.0"?>

<launch>

	<!-- transform between checkerboard coordinate system and checkerboard_reference coordinate system
		checkerboard: starts in upper left corner of checkerboard pattern, x goes horizontal to the right, y goes downwards to gravity direction, z goes into the plane
		checkerboard_reference: similar definition as checkerboard, but located on the left edge of the box in front of the wall at the height of the laser scanner
		transform between them: no rotation, just a x and y offset (ATTENTION: y offset depends on mounting height of the laser scanner) -->
	<node pkg="tf" type="static_transform_publisher" name="static_transform_publisher_checkerboard_reflector" output="screen" args="0.0 -0.7 0 0 0 0  checkerboard_reference checkerboard 100"/>
	<node pkg="tf" type="static_transform_publisher" name="static_transform_publisher_checkerboard_reflector_nav" output="screen" args="0.0 0.0 0.0 -1.57079632679 0 -1.57079632679  checkerboard_reference_nav checkerboard_reference 100"/>

	<node ns="checkerboard_localization" name="relative_localization" pkg="relative_localization" type="relative_localization_node" output="screen">
		<rosparam command="load" file="$(find relative_localization)/ros/launch/checkerboard_localization_params.yaml"/>
		<param name="localization_method" value="checkerboard" />
		<remap from="laser_scan_in" to="/scan" />
	</node>
</launch>
